xbmc/ServiceManager.cpp

```diff
@@ -153,6 +153,9 @@ bool CServiceManager::InitStageTwo(const CAppParamParser &params)
 
   m_gameRenderManager.reset(new RETRO::CGUIGameRenderManager);
 
+  // we need to bump init_level before the file extension provider is created
+  // or it will not obtain add-on extensions.
+  init_level = 2;
```

Comment:
Should we introduce an independent state flag for the binary cache and set it true if the cache is ready and check it in `IsBinaryCacheUp`?